pkg/stanza/operator/input/syslog/syslog.go

```diff
@@ -136,3 +142,40 @@ func (t *Input) SetOutputs(operators []operator.Operator) error {
 	t.parser.SetOutputIDs(t.GetOutputIDs())
 	return t.parser.SetOutputs(operators)
 }
+
+func OctetMultiLineBuilder() (bufio.SplitFunc, error) {
+	return newOctetFrameSplitFunc(true), nil
+}
+
+func newOctetFrameSplitFunc(flushAtEOF bool) bufio.SplitFunc {
+	frameRegex := regexp.MustCompile(`^[1-9]\d*\s`)
+	return func(data []byte, atEOF bool) (advance int, token []byte, err error) {
+		frameLoc := frameRegex.FindIndex(data)
+		if frameLoc == nil {
+			// Flush if no more data is expected
+			if len(data) != 0 && atEOF && flushAtEOF {
+				token = data
+				advance = len(data)
+				return
+			}
+			return 0, nil, nil
+		}
+
+		frameMaxIndex := frameLoc[1]
+		// delimit space between length and log
+		frameLenValue, err := strconv.Atoi(string(data[:frameMaxIndex-1]))
+		if err != nil {
+			return 0, nil, err // read more data and try again.
+		}
+
+		advance = frameMaxIndex + frameLenValue
+		// the limitation here is that we can only line split within a single buffer
+		// the context of buffer length cannot be pass onto the next scan
+		if advance > cap(data) {
+			return 0, nil, errors.New("frame size is larger than buffer capacity")
+		}
```

Comment:
Would it be better to just advance to the end of the buffer?

Reply:
it is to continue data flowing. let me make that change and update tests